File: OCH_Tao/ABC/A/097.js
const [A, B, C, D] = ((require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"))[0].split(" ")).map(x => parseInt(x, 10));
if (Math.abs(A - C) <= D) {
  console.log("Yes");
} else if (Math.abs(A - B) <= D && Math.abs(B - C) <= D) {
  console.log("Yes");
} else {
  console.log("No");
}

File: OCH_Tao/ABC/A/097.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
  int A, B, C, D;
  cin >> A >> B >> C >> D;
  if (abs(A - C) <= D || (abs(A - B) <= D && abs(B - C) <= D))
  {
    cout << "Yes" << endl;
  }
  else
  {
    cout << "No" << endl;
  }
}